### ISSUE TITLE
🔥A second initialized app takes options from the default one

### ISSUE THREAD

**Issue by cbrwizard** [invertase/react-native-firebase#1732]
Hello, it seems that the second initialized app is taking options from the json/plist Google files instead of using the values I provide. I'm following the tutorial https://rnfirebase.io/docs/v5.x.x/core/initialize-apps

```
// fake values
  const iosConfig = {
    apiKey: 'key',
    appId: 'Id',
    databaseURL: 'url',
    messagingSenderId: 'sender',
    projectId: 'staging',
    storageBucket: 'bucket',
  }
  const app = await firebase.initializeApp(iosConfig, 'dev').onReady()
  console.log({ initializedDevApp: app }) // has a different name `dev`, but the values in `options` and services like `database` are taken from files. I expected them to be taken from the config I provided above.
  console.log({ defaultApp: firebase.app() }) // has a name `default` and correct, expected values
```
The goal is to have 2 apps initialized with different configs: one for staging, one for dev. I need to dynamically switch between them.
Maybe I'm doing it wrong? Please help.

## Issue

---

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'MothershipCompanion' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Required by RNFirebase
  pod 'Firebase/Core', '~> 5.11.0'

  # https://rnfirebase.io/docs/v5.x.x/crashlytics/ios
  pod 'Fabric', '~> 1.7.13'
  pod 'Crashlytics', '~> 3.10.7'

  # https://rnfirebase.io/docs/v5.x.x/perf-mon/ios#Add-the-pod
  pod 'Firebase/Performance', '~> 5.11.0'

  # https://rnfirebase.io/docs/v5.x.x/database/ios#Add-the-pod
  pod 'Firebase/Database', '~> 5.11.0'

  # Pods for MothershipCompanion

  platform :ios, '9.0'
  target 'MothershipCompanionTests' do
    inherit! :search_paths
    # Pods for testing

      # https://facebook.github.io/react-native/docs/integration-with-existing-apps
      # Your 'node_modules' directory is probably in the root of your project,
      # but if not, adjust the `:path` accordingly
      pod 'React', :path => '../node_modules/react-native', :subspecs => [
        'Core',
        'CxxBridge', # Include this for RN >= 0.47
        'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
        'RCTText',
        'RCTNetwork',
        'RCTWebSocket', # Needed for debugging
        'RCTAnimation', # Needed for FlatList and animations running on native UI thread
        # Add any other subspecs you want to use in your project
      ]
      # Explicitly include Yoga if you are using RN >= 0.42.0
      pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

      # Third party deps podspec link
      pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
      pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
      pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  end

end

```

#### `AppDelegate.m`:

```objc
/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"
#import <CodePush/CodePush.h>
#import <Firebase.h>

#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>
#if __has_include(<React/RNSentry.h>)
#import <React/RNSentry.h> // This is used for versions of react >= 0.40
#else
#import "RNSentry.h" // This is used for versions of react < 0.40
#endif

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  [FIRApp configure]; // https://rnfirebase.io/docs/v5.x.x/installation/ios

  // https://github.com/Microsoft/react-native-code-push/blob/master/docs/setup-ios.md#plugin-configuration-ios
  #ifdef DEBUG
//    jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios&dev=true"];
    jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
  #else
    jsCodeLocation = [CodePush bundleURL];
  #endif
RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"MothershipCompanion"
                                               initialProperties:nil
                                                   launchOptions:launchOptions];

  [RNSentry installWithRootView:rootView];

  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  return YES;
}

@end

```

---

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

---

## Environment

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [X] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **If known, the version of the platform are you experiencing the issue on:**
  - iOS 9
- **Operating System:**
  - [X] MacOS, version: `10.14`
  - [ ] Windows, version: `N/A`
  - [ ] Other, please specify: `N/A`
- **Build Tools:**
  - `Xcode 10`
- **`React Native` version:**
  - `0.57.0`
- **`React Native Firebase` library version:**
  - `5.1.1`
- **`Firebase` module(s) you're using that has the issue:**
  - [x] **N/A**
  - [ ] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [ ] Cloud **Messaging** (FCM)
  - [ ] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] Invites
  - [ ] Instance ID
  - [ ] Notifications
  - [ ] Performance Monitoring
  - [ ] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [ ] No
  - [X] Yes, version: `3.0.1`
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](./../../CONTRIBUTING.md)

**Comment by cbrwizard** [comment]
I investigated this and looks like something inside firebase internally caches the second app's settings.

Look at this screenshot. The only change in code between them is that I renamed the firebase app from 'deve' to 'cock'.

As a solution in my case, I've set the app name to be the same as project id; that way I'm not bothered by the cache. 
<img width="1608" alt="screenshot 2019-01-07 at 17 23 29" src="https://user-images.githubusercontent.com/1615659/50773553-31bd4400-12a2-11e9-9eff-c3965538d927.png">

That is indeed weird and led to many hours spent on debugging. If that's in the repo's code consider changing that or explicitly stating the behavior.
